src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -1,201 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System.Diagnostics.CodeAnalysis;
-using System.IO;
-using System.Text;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Internal.Log;
-using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Host
-{
-    internal abstract partial class AbstractSyntaxTreeFactoryService
-    {
-        internal readonly struct SyntaxTreeInfo
-        {
-            public readonly string FilePath;
-            public readonly ParseOptions Options;
-            public readonly ITextAndVersionSource TextSource;
-            public readonly LoadTextOptions LoadTextOptions;
-            public readonly Encoding Encoding;
-            public readonly int Length;
-            public readonly bool ContainsDirectives;
```

Comment:
@jasonmalinowski see conversation below about losign the fast access to things like Length/ContainsDirective. We may want to consider an alternative approach to intelligently assocaite this with a tree.